packages/terra-overlay/src/Overlay.jsx

```diff
@@ -134,21 +142,24 @@ class Overlay extends React.Component {
     }
   }
 
-  resetBeforeOverlay() {
+  resetBeforeOverlay(selector) {
     document.documentElement.style.overflow = this.overflow;
+    if (document.querySelector(selector)) {
```

Comment:
Let's also add this conditional check inside a componentWillUnMount lifecycle method.